sys-cluster/mpich/mpich-3.2-r1.ebuild

```diff
@@ -0,0 +1,136 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=5
+
+FORTRAN_NEEDED=fortran
+
+inherit fortran-2 multilib-minimal
+
+MY_PV=${PV/_/}
+DESCRIPTION="A high performance and portable MPI implementation"
+HOMEPAGE="http://www.mpich.org/"
+SRC_URI="http://www.mpich.org/static/downloads/${PV}/${P}.tar.gz"
+
+SLOT="0"
+LICENSE="mpich"
+KEYWORDS=""
+IUSE="+cxx doc fortran mpi-threads romio threads"
+
+COMMON_DEPEND="
+	>=dev-libs/libaio-0.3.109-r5[${MULTILIB_USEDEP}]
+	>=sys-apps/hwloc-1.10.0-r2[${MULTILIB_USEDEP}]
+	romio? ( net-fs/nfs-utils )"
+
+DEPEND="${COMMON_DEPEND}
+	dev-lang/perl
+	sys-devel/libtool"
+
+RDEPEND="${COMMON_DEPEND}
+	!sys-cluster/mpich2
+	!sys-cluster/openmpi"
+
+S="${WORKDIR}"/${PN}-${MY_PV}
+
+MULTILIB_WRAPPED_HEADERS=(
+	/usr/include/mpicxx.h
+	/usr/include/mpi.h
+	/usr/include/opa_config.h
+)
+
+pkg_setup() {
+	FORTRAN_STANDARD="77 90"
+	fortran-2_pkg_setup
+
+	if use mpi-threads && ! use threads; then
+		ewarn "mpi-threads requires threads, assuming that's what you want"
```

Comment:
Imo, this is too much implicit inference. We should rather go for `REQUIRED_USE="mpi-threads? ( threads )"`, and then we can chuck this whole `pkg_setup()` function.